roq-core/src/main/java/org/roqmessaging/core/factory/RoQConnectionFactory.java

```diff
@@ -39,34 +44,68 @@
 	//The socket to the global config
 	private Socket globalConfigReq;
 	
+	// TODO: Following params in a config file
+	// Number of times that the client retry the request
+	private int maxRetry = 10;
+	// the rcv timeout of ZMQ
+	private int timeout = 5000;
+	private RoQZooKeeper zk;
 	private Logger logger = Logger.getLogger(RoQConnectionFactory.class);
 	
 	/**
 	 * Build  a connection Factory and takes the location of the global configuration manager as input
-	 * @param configManager the config manager IP address, the default port 5000 will be applied
+	 * @param configManager the zookeeper IP addresses, the default port 5000 will be applied
+	 * @param maxRetry the number of times that the client try to process the request
+	 * @param the timeout between each retry
+	 */
+	public RoQConnectionFactory(String zkAddresses, int maxRetry, int timeout) {
+		zk = new RoQZooKeeper(zkAddresses);
+		zk.start();
+		this.maxRetry = maxRetry;
+		this.timeout = timeout;
+	}
+	
+	/**
+	 * Build  a connection Factory and takes the location of the global configuration manager as input
+	 * @param configManager the zookeeper IP addresses, the default port 5000 will be applied
+	 */
+	public RoQConnectionFactory(String zkAddresses) {
+		zk = new RoQZooKeeper(zkAddresses);
+		zk.start();
+	}
+
+	/**
+	 * Build  a connection Factory and takes the location of the global configuration manager as input
+	 * @param cfg Zookeeper config
+	 * @param maxRetry the number of times that the client try to process the request
+	 * @param the timeout between each retry
 	 */
-	public RoQConnectionFactory(String configManager) {
-		this.configServer= configManager;
-		
+	public RoQConnectionFactory(RoQZKSimpleConfig cfg) {
```

Comment:
This specific constructor does not look like it is in use. If that is the case, you can remove it. You can probably keep only one of the three constructors here.